webapp/channels/src/packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -433,3 +446,76 @@ export function decrementThreadCounts(post: ExtendedPost) {
         });
     };
 }
+
+export function getThreadsForChannel(channelId: string, {before = '', after = '', perPage = ThreadConstants.THREADS_CHUNK_SIZE, totalsOnly = false, threadsOnly = false, extended = false, since = 0, filter}: FetchChannelThreadOptions = {}) {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let data: undefined | ChannelThreadList;
+
+        try {
+            data = await Client4.getThreadsForChannel(channelId, {before, after, perPage, extended, totalsOnly, threadsOnly, since, filter});
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        if (!data) {
+            return {error: true};
+        }
+
+        if (data?.threads?.length) {
+            await dispatch(getMissingProfilesByIds(uniq(data.threads.map(({participants}) => participants.map(({id}) => id)).flat())));
+
+            dispatch({
+                type: PostTypes.RECEIVED_POSTS,
+                data: {posts: data.threads.map(({post}) => ({...post, update_at: 0}))},
```

Comment:
Coming back to https://github.com/mattermost/mattermost-server/pull/22928/files/fd56c07d9a49b2e952fc8685fd9d656972269aaf#r1173678473 and https://github.com/mattermost/mattermost-server/pull/22928/files/fd56c07d9a49b2e952fc8685fd9d656972269aaf#r1173112049, this isn't something urgent since we've been doing it for quite a while and it doesn't seem to have caused any major issues, but the fact that we're not getting metadata for these posts has me a bit worried we might run into weird cases because it breaks an assumption we've had for a long time that we load everything needed to render a post at once. For example, this might be why I've been seeing "scroll pop" issues reoccurring, but I haven't looked into that enough to say.

As for the threads APIs being heavy, this definitely helps to make them a bit lighter, but then we might end up immediately calling the API multiple times to get files from individual threads, so instead of one heaver `getThreadsForChannel` call, we have a slightly lighter one followed by up to `THREADS_CHUNK_SIZE` `getFIleInfosForPost` ones. That was another motivator for adding post metadata in the first place.

That all said, since that doesn't seem to be causing issues with the current code, I feel like we can probably just keep an eye on this. Worst case scenario, we could always look at storing these posts in a separate part of Redux or wait until we have a better way to load all that data so that we're not forced to load it all up front.